rfcs/0088-commit-message-guideline.md

```diff
@@ -0,0 +1,181 @@
+for new contributions because much more time will be necessary trying to
+understand what motivated a past critical but unclear change.
+
+Hence this Commit Message Guideline can help contributors to write good commit
+messages and so improve the current situation regarding the TVM commit logs.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Commit Message Guideline
+
+Apache TVM uses the Github (GH) platform for patch submission and code review
+via Pull Requests (PRs). The final commit (title and body) that is merged into
+the Apache TVM main tree is composed of the PR's title and body and must be kept
+updated and reflecting the new changes in the code as per the reviews and
+discussions.
+
+Although these guidelines apply essentially to the PRs’ title and body messages,
+because GH auto-generates the PR’s title and body from the commits on a given
+branch, it’s recommended to follow these guidelines right from the beginning,
+when preparing commits in general to be submitted to the Apache TVM project.
+This will ease the creation of a new PR, avoiding rework, and also will help the
+review.
+
+The rules below will help to achieve uniformity that has several benefits, both
+for review and for the code base maintenance as a whole, helping you to write
+commit messages with a good quality suitable for the Apache TVM project,
+allowing fast log searches, bisecting, and so on.
+
+_PR/commit title_:
+
+* Guarantee a title exists (enforced);
+* Don’t use Github usernames in the title, like @username (enforced);
+* Check if a tag should be present as a hint about what component(s) of the code
+  the commits “touch”. For example [BugFix], [CI], [microTVM], and [TVMC]. Tags
+  go between square brackets and appear first in the title. Tags help reviewers
+  to identify the PRs they can/want to review and also help the release folks
+  when generating the release notes (enforced);
+* Use an imperative mood. Avoid titles like “Added operator X” and “Updated
+  image Y in the CI”, instead use the forms “Add feature X” and “Update image Y
+  in the CI” instead;
+* Observe proper use of caps at the beginning (uppercase for the first letter)
+  and for acronyms, like, for instance, TVM, FVP, OpenCL. Hence instead of
+  “fix tvm use of opencl library”, write it as “Fix TVM use of OpenCL library”;
+* No period at the end of the title is necessary.
+
+_PR/commit body_:
+
+* Guarantee a body exists (enforced);
+* Don’t use Github usernames in body text, like @username (enforced);
+* Avoid “bullet” commit message bodies: “bullet” commit message bodies are not
+  bad per se, but “bullet” commit messages without any description or
+  explanation is likely as bad as commits without any description, rationale,
+  or explanation in the body.
+
+For minor deviations from these guidelines, the community will normally favor
+reminding the contributor of this policy over reverting or blocking a commmit /
+PR.
```

Comment:
I'm not sure. As committer I agree to fixing minor deviations (a period in the end of a commit title, for instance), that's ok. However ignoring deviations is not fine, ignoring _minor_  deviations is fine. The enforced rules are not minor deviations, having a PR without a body (which will result in a commit message in the main tree without a body) is not a minor deviation. On the other hand a period in the end  of the commit title is a minor deviation, but what else can be considered a minor deviation? I'm certainly appealing to the community's good faith and will when interpreting this part. In general I would prefer the committer -- if she/he so wishes -- to let the minor deviations pass and **remind** the author, so helping to educate the contributors, than having committers spending cycles fixing such a minor deviations.

Reply:
I think given part of the committing process involves the committer copying the title/body of the PR, it's not unreasonable for them to also add a period at that stage and would add no time to the review. Obviously if this gets forgotten it's not the end of the world, but I think the guidance should be in favour of fixing minor issues rather than leaving it up to interpretation.

Reply:
@mbaret I see. Would adding "Committers are encouraged to fix those minor deviations before merging the PR" work? If not, please suggest an alternative :)

Reply:
@mbaret friendly ping.